### PR TITLE
tgl-006-drop-stable: port DSM and DMIC patches from master

### PR DESCRIPTION
**Pull request by RanderWang**
(1) enable DSM feature for sdw
(2) fix DMIC ipc timeout issues

### CODE REVIEW

**Comment by RanderWang** [comment]
@ClarexZhou  this PR is for 006 release, thanks!

File: tools/topology/sof-tgl-sdw-max98373-rt5682.m4 [hunk 1]
```diff
@@ -158,7 +161,7 @@ DAI_ADD(sof/pipe-dai-capture.m4,
 # Buffers use s16le format, with 16 frame per 1000us on core 0 with priority 0
 DAI_ADD(sof/pipe-dai-capture.m4,
         6, DMIC, 1, dmic16k,
-        PIPELINE_SINK_6, 2, s16le,
+        PIPELINE_SINK_6, 2, s32le,
```
Comment (keyonjie) [inline tools/topology/sof-tgl-sdw-max98373-rt5682.m4:164]: update the comment above also.

Comment (RanderWang) [reply tools/topology/sof-tgl-sdw-max98373-rt5682.m4:164]: done

**Comment by jajanusz** [comment]
@mengdonglin @RanderWang 
Do we really need this on 006? It's already commited release, better to just have it on 007 imho

**Comment by RanderWang** [comment]
not used

**Pull request closed without merging**